src/main/java/org/jenkinsci/plugins/GithubSecurityRealm.java

```diff
@@ -693,17 +694,17 @@ public UserDetails loadUserByUsername(String username)
 
         Authentication token = SecurityContextHolder.getContext().getAuthentication();
 
-        if (token == null || !username.equals(token.getPrincipal())) {
-            if(localUser != null && GithubSecretStorage.contains(localUser)){
+        try {
+            if(localUser != null && GithubSecretStorage.contains(localUser)) {
                 String accessToken = GithubSecretStorage.retrieve(localUser);
-                try {
-                    token = new GithubAuthenticationToken(accessToken, getGithubApiUri());
-                } catch (IOException e) {
-                    throw new UserMayOrMayNotExistException("Could not connect to GitHub API server, target URL = " + getGithubApiUri(), e);
-                }
-                SecurityContextHolder.getContext().setAuthentication(token);
```

Comment:
This part is what kept throwing the CSRF errors because global security page would generate a new CSRF token for the authenticated user every time the page was visited.  So the form on the page would have an out-of-date CSRF token since doCheck would generate a new one for the logged in user (admin).